mm2src/mm2_main/src/rpc/lp_commands/lp_commands_legacy.rs

```diff
@@ -33,49 +33,117 @@ use std::borrow::Cow;
 
 use crate::mm2::lp_dispatcher::{dispatch_lp_event, StopCtxEvent};
 use crate::mm2::lp_network::subscribe_to_topic;
-use crate::mm2::lp_ordermatch::{cancel_orders_by, CancelBy};
+use crate::mm2::lp_ordermatch::{cancel_orders_by, get_matching_orders, CancelBy};
 use crate::mm2::lp_swap::{active_swaps_using_coin, tx_helper_topic, watcher_topic};
 use crate::mm2::MmVersionResult;
 
+const INTERNAL_SERVER_ERROR_CODE: u16 = 500;
+const RESPONSE_OK_STATUS_CODE: u16 = 200;
+
 /// Attempts to disable the coin
 pub async fn disable_coin(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, String> {
     let ticker = try_s!(req["coin"].as_str().ok_or("No 'coin' field")).to_owned();
-    let _coin = match lp_coinfind(&ctx, &ticker).await {
+    let coin = match lp_coinfind(&ctx, &ticker).await {
         Ok(Some(t)) => t,
         Ok(None) => return ERR!("No such coin: {}", ticker),
         Err(err) => return ERR!("!lp_coinfind({}): ", err),
     };
-    let swaps = try_s!(active_swaps_using_coin(&ctx, &ticker));
-    if !swaps.is_empty() {
-        let err = json!({
-            "error": format!("There're active swaps using {}", ticker),
-            "swaps": swaps,
-        });
-        return Response::builder()
-            .status(500)
-            .body(json::to_vec(&err).unwrap())
-            .map_err(|e| ERRL!("{}", e));
+    let coins_ctx = CoinsContext::from_ctx(&ctx).map_err(|err| ERRL!("{}", err))?;
+
+    // Get all enabled tokens with platform coin including the coin.
+    let coins_to_disable = coins_ctx
+        .get_tokens_to_disable(&ticker)
+        .await
+        .into_iter()
+        .chain(std::iter::once(coin.platform_ticker().to_string()))
+        .collect::<Vec<_>>();
+
+    // NOTE. We'll skip this step if we are not disabling a platform coin!.
+    // If coins_to_disable list is > 1 then it's a platform and it tokens we want to disable hence, we check if
+    // there's an active or matching swaps before we try disabling the token.
+    if coins_to_disable.len() > 1usize {
+        for ticker in &coins_to_disable {
+            let swaps = try_s!(active_swaps_using_coin(&ctx, ticker));
+            let still_matching = get_matching_orders(&ctx, ticker).await;
+
+            if !still_matching.is_empty() || !swaps.is_empty() {
+                let err = json!({
+                    "error": format!("There're currently matching orders or active swaps using {}", ticker),
+                    "orders": {
+                        "active_swaps": swaps,
+                        "still_matching": still_matching
+                    }
+                });
```

Comment:
Please return all the active swaps and still matching orders for all tokens and for the platform coin in the error. Right now, you return an error for the first found token with an active swap and/or a still matching orders.

Reply:
done